File: backend/src/main/java/protasks/backend/Board/BoardUsersPermId.java
package protasks.backend.Board;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import java.io.Serializable;
import java.util.Objects;

@Embeddable
public class BoardUsersPermId implements Serializable {

    @Column(name = "board_id")
    private Long boardId;

    @Column(name = "user_id")
    private Long userId;

    public BoardUsersPermId() {}

    public BoardUsersPermId(
            Long boardId,
            Long userId) {
        this.boardId = boardId;
        this.userId = userId;
    }

    public Long getBoardId() {
        return boardId;
    }

    public void setBoardId(Long boardId) {
        this.boardId = boardId;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;

        if (o == null || getClass() != o.getClass())
            return false;

        BoardUsersPermId that = (BoardUsersPermId) o;
        return Objects.equals(boardId, that.boardId) &&
                Objects.equals(userId, that.userId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(boardId, userId);
    }
}

File: app/src/main/java/com/example/protasks/models/BoardUsersPermId.kt
package com.example.protasks.models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class BoardUsersPermId(
    @Expose
    @SerializedName("boardId")
    private var boardId: Long?, @Expose
    @SerializedName("userId")
    private var userId: Long?
) {

    fun getBoardId(): Long? {
        return boardId
    }

    fun setBoardId(boardId: Long?) {
        this.boardId = boardId
    }

    fun getUserId(): Long? {
        return userId
    }

    fun setUserId(userId: Long?) {
        this.userId = userId
    }

}